ush/load_ufsda_modules.sh

```diff
@@ -27,43 +27,42 @@ fi
 ulimit_s=$( ulimit -S -s )
 
 # Find module command and purge:
+source "${HOMEgfs}/ush/detect_machine.sh"
 source "${HOMEgfs}/ush/module-setup.sh"
 
 # Load our modules:
 module use "${HOMEgfs}/sorc/gdas.cd/modulefiles"
 
-if [[ -d /lfs/f1 ]]; then
+if [[ "${MACHINE_ID}" == "wcoss2" || "${MACHINE_ID}" == "acorn" ]]; then
   # We are on WCOSS2 (Cactus or Dogwood)
   echo WARNING: UFSDA NOT SUPPORTED ON THIS PLATFORM
-elif [[ -d /lfs3 ]] ; then
+elif [[ "${MACHINE_ID}" == "jet" ]] ; then
   # We are on NOAA Jet
   echo WARNING: UFSDA NOT SUPPORTED ON THIS PLATFORM
-elif [[ -d /scratch1 ]] ; then
+elif [[ "${MACHINE_ID}" == "hera" ]] ; then
   # We are on NOAA Hera
   module load "${MODS}/hera"
   # set NETCDF variable based on ncdump location
-  NETCDF=$( which ncdump )
+  NETCDF=$( command -v ncdump)
   export NETCDF
-elif [[ -d /work ]] ; then
-  # We are on MSU Orion or Hercules
-  if [[ -d /apps/other ]] ; then
-     # Hercules
-     module load "${MODS}/hercules"
-  else
-     # Orion
-     module load "${MODS}/orion"
-  fi
+elif [[ "${MACHINE_ID}" == "orion" ]] ; then
+  # We are on MSU Orion
+  module load "${MODS}/orion"
   # set NETCDF variable based on ncdump location
-  ncdump=$( which ncdump )
+  ncdump=$( command -v ncdump)
   NETCDF=$( echo "${ncdump}" | cut -d " " -f 3 )
   export NETCDF
-elif [[ -d /glade ]] ; then
-  # We are on NCAR Yellowstone
-  echo WARNING: UFSDA NOT SUPPORTED ON THIS PLATFORM
-elif [[ -d /lustre && -d /ncrc ]] ; then
+elif [[ "${MACHINE_ID}" == "hercules" ]] ; then  
+  # We are on MSU Hercules
+  module load "${MODS}/hercules"
+  # set NETCDF variable based on ncdump location
+  ncdump=$( command -v ncdump )
+  NETCDF=$( echo "${ncdump}" | cut -d " " -f 3 )
+  export NETCDF
```

Comment:
For orion and hercules, this can also be collected:
```bash
case "${MACHINE_ID}" in
  "orion" | "hercules")
    ncdump=$( command -v ncdump)
    NETCDF=$( echo "${ncdump}" | cut -d " " -f 3 )
    export NETCDF
    ;;
  *)
    echo "NetCDF does not need special handling # TODO: what is this for?
    ;;
esac
```